### DISCUSSION TITLE
submission ERROR: run timeout(>30000)

### DISCUSSION THREAD

**Discussion by yuhao-morning** [PaddlePaddle/PaddleSpatial#111] [BaiduKDDCup2022]
In my local environment, predicting 134 turbs only takes 1 hour and an half. Buy why online it has this error? Can you check the backend log for me?

**Comment by zhoujingbo** [comment]
Can you provide you team name and submission time?

**Reply by yuhao-morning** [reply]
Yesterday it doesn't have this error, but error becomes "list index out of range". submission start: 05-27 17:08. Again, in my local environment everything is OK, it gives the result.

**Comment by yuhao-morning** [comment]
yu_1982_lyTeam  
submit time:  yesterday 15:45



---- Replied Message ----
| From | zhou ***@***.***> |
| Date | 05/27/2022 17:05 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [PaddlePaddle/PaddleSpatial] submission ERROR: run timeout(>30000) (Discussion #111) |

Can you provide you team name and submission time?

**Comment by yuhao-morning** [comment]
Is there any log released that you can let me know about this issue? 
team name: yh_1982_lyTeam
submit start 2022-05-27 17:08

**Reply by ramber1836** [reply]
![image](https://user-images.githubusercontent.com/19561583/170850384-60418a5c-892b-499a-a092-c7f13ef65859.png)

Please avoid unnecessary details being printed in your code, which can make the debug easier to locate the problem.

**Reply by yuhao-morning** [reply]
Have removed unnecessary printing. Can you check again:
Team: yh_1982_lyTeam
submit start: 05-30 09:45

**Comment by yuhao-morning** [comment]
Have removed unnecessary printing. Can you check again:
Team: yh_1982_lyTeam
submit start: 05-30 09:45

**Reply by zhoujingbo** [reply]
<img width="932" alt="Xnip2022-05-30_23-52-45" src="https://user-images.githubusercontent.com/6733782/171027353-183723e4-110b-4d41-bdb6-8e79a211d7b8.png">

It still is a list index out of range problem

**Reply by yuhao-morning** [reply]
Is you test_data updated recently or evaluation.py updated ? why in my local environment, everything is OK?

**Reply by yuhao-morning** [reply]
why do you have a NUM_MAX_RUNS in your updated evaluation.py? That is the reason I find that causes"list index out of range". Isn't it that just 1 ? Why did you set it 195? Are there any 195 files in test_x.zip or test_y.zip ?

**Reply by ramber1836** [reply]
I don't think it is the reason cause your code 'list index out of range' .
This variable is in accordance with the number of test files, i.e. how many predictions are in the online test.

**Comment by ramber1836** [comment]
Please update your local GitHub repo from time to time. The evaluation code in the GitHub repo is in accordance with the online test.

**Reply by yuhao-morning** [reply]
can you check this one?
start from: 05-31 11:53

**Comment by yuhao-morning** [comment]
start from: 05-31 11:53 
this one has been evaluating for hours. Can you check that?

**Reply by ramber1836** [reply]
When the submitted code is running, the backend log can only be accessible to the system administrators. So we cannot check the backend log unless the evaluation is finished. 
BTW, I checked the status of your submission, it has been terminated without returning the backend log. The probable reason is that your evaluation is out of time.

**Reply by yuhao-morning** [reply]
After nearly 10 hours evaluation, it throws the error '异常'. What is the reason? Is the log available now? In my local , for one test data, it runs just about  only 1 mins. I am sure it is not that the running time is long, completely not .

**Reply by ramber1836** [reply]
Here is a workaround solution: you can raise an exception in your code after 2 runs, then we can check the log of your submission accordingly to see what was going wrong. 
Besides, try to avoid unnecessary stdout as much as you can in your code, which is really not helpful at all. 

Shown below, the messy stdout in your previous submission at 2022-05-31 00:39:48 is not informative: 
![image](https://user-images.githubusercontent.com/19561583/171208840-45927fca-1f20-41fa-bf91-a6d354467ac6.png)

And in this submission, your model takes more than 2 hours for one prediction: 
![image](https://user-images.githubusercontent.com/19561583/171209723-5520d463-a772-4937-a73d-53655a783f73.png)

**Reply by yuhao-morning** [reply]
how about 05-31 23:03.? It is "异常" again.

this is msg in my local:
Elapsed time for 0-th prediction is: 16.2078914642334 secs 

tmp dir 2  /tmp/tmpoxi3cqy3
path to data  /data/yuhao/Autoformer/test_y.zip
files  ['0001out.zip']

	The 0-th prediction -- RMSE: 59.09615556137181, MAE: 42.06601748712154, Score: 50.58108652424667, and Accuracy: 48.1750%

**Reply by ramber1836** [reply]
> how about 05-31 23:03.? It is "异常" again.
> 
> this is msg in my local: Elapsed time for 0-th prediction is: 16.2078914642334 secs
> 
> tmp dir 2 /tmp/tmpoxi3cqy3 path to data /data/yuhao/Autoformer/test_y.zip files ['0001out.zip']
> 
> ```
> The 0-th prediction -- RMSE: 59.09615556137181, MAE: 42.06601748712154, Score: 50.58108652424667, and Accuracy: 48.1750%
> ```

No log.
As I mentioned before, if the evaluation is terminated by the system due to some reasons, like an out-of-time error, no log is available for us. 
We are working on it to avoid the issue that the evaluation is terminated without generating any logs.

**Comment by yuhao-morning** [comment]
May I know if the packages:
SIX, 

scipy.sparse.linalg




are installed online?
















At 2022-06-01 10:17:35, "Xinjiang Lu" ***@***.***> wrote:

how about 05-31 23:03.? It is "异常" again.

this is msg in my local: Elapsed time for 0-th prediction is: 16.2078914642334 secs

tmp dir 2 /tmp/tmpoxi3cqy3 path to data /data/yuhao/Autoformer/test_y.zip files ['0001out.zip']

The 0-th prediction -- RMSE: 59.09615556137181, MAE: 42.06601748712154, Score: 50.58108652424667, and Accuracy: 48.1750%



No log either.
As I mentioned before, if the evaluation is terminated by the system due to some reasons, like an out-of-time error, no log is available for us.
We are working on it to avoid the issue that the evaluation is terminated without generating any logs.

**Comment by yuhao-morning** [comment]
Can you check  log again:

Team :yh_1982_lyTeam
submit time: 2022-07-03 19:43





May I know if the packages:
SIX, 

scipy.sparse.linalg




are installed online?
















At 2022-06-01 10:17:35, "Xinjiang Lu" ***@***.***> wrote:

how about 05-31 23:03.? It is "异常" again.

this is msg in my local: Elapsed time for 0-th prediction is: 16.2078914642334 secs

tmp dir 2 /tmp/tmpoxi3cqy3 path to data /data/yuhao/Autoformer/test_y.zip files ['0001out.zip']

The 0-th prediction -- RMSE: 59.09615556137181, MAE: 42.06601748712154, Score: 50.58108652424667, and Accuracy: 48.1750%



No log either.
As I mentioned before, if the evaluation is terminated by the system due to some reasons, like an out-of-time error, no log is available for us.
We are working on it to avoid the issue that the evaluation is terminated without generating any logs.

**Reply by ramber1836** [reply]
Traceback (most recent call last):
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 332, in eval
    return evaluate(tmp_dir)
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 233, in evaluate
    pred_res = predict_and_test(envs, PRED_DIR, forecast_module, i, flag='predict')
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 168, in predict_and_test
    return exec_predict_and_test(envs, path_to_test_file, forecast_module, flag)
  File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 131, in exec_predict_and_test
    "prediction": forecast_module.forecast(envs)
  File "/tmp/tmpnju2ftxg/kdd2022_submit_FEDformer/predict.py", line 204, in forecast
  File "/tmp/tmpnju2ftxg/kdd2022_submit_FEDformer/models/FEDformer.py", line 23, in __init__
AttributeError: 'dict' object has no attribute 'version'

**Comment by yuhao-morning** [comment]
Exception: Different shapes between Prediction and Ground Truth, shape of Ground Truth: (144,), shape of Prediction: (134, 1)


Why the shape of groundtruth is (144,)? Isn't that the number of wind Turb 134?
















At 2022-07-04 10:58:58, "Xinjiang Lu" ***@***.***> wrote:

Traceback (most recent call last):
File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 332, in eval
return evaluate(tmp_dir)
File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 233, in evaluate
pred_res = predict_and_test(envs, PRED_DIR, forecast_module, i, flag='predict')
File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 168, in predict_and_test
return exec_predict_and_test(envs, path_to_test_file, forecast_module, flag)
File "/home/aistudio/eval_prog/kddcup22-sdwpf-evaluation/pytorch/evaluation.py", line 131, in exec_predict_and_test
"prediction": forecast_module.forecast(envs)
File "/tmp/tmpnju2ftxg/kdd2022_submit_FEDformer/predict.py", line 204, in forecast
File "/tmp/tmpnju2ftxg/kdd2022_submit_FEDformer/models/FEDformer.py", line 23, in init
AttributeError: 'dict' object has no attribute 'version'

**Reply by ramber1836** [reply]
There are 134 turbines.
Please provide your teamname and submission time so that I can check the detailed error log.

**Comment by yuhao-morning** [comment]
I get an error for importing paddle:


import paddle




ImportError: cannot import name '_non_static_mode' from 'paddle.fluid.framework' (/opt/conda/envs/python35-paddle120-env/lib/python3.7/site-packages/paddle/fluid/framework.py)


Can you help me ? 
















At 2022-07-11 10:31:47, "Xinjiang Lu" ***@***.***> wrote:

There are 134 turbines.
Please provide your teamname and submission time so that I can check the detailed error log.

**Comment by yuhao-morning** [comment]
Is there any result? Can  the log be released   ?



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2022年05月27日 17:33 |
| 收件人 | ***@***.***> |
| 抄送至 | |
| 主题 | Re: [PaddlePaddle/PaddleSpatial] submission ERROR: run timeout(>30000) (Discussion #111) |
yu_1982_lyTeam  
submit time:  yesterday 15:45



---- Replied Message ----
| From | zhou ***@***.***> |
| Date | 05/27/2022 17:05 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [PaddlePaddle/PaddleSpatial] submission ERROR: run timeout(>30000) (Discussion #111) |

Can you provide you team name and submission time?
